src/fsharp/FSharp.Core/array.fs

```diff
@@ -494,12 +494,23 @@ namespace Microsoft.FSharp.Collections
 
         [<CompiledName("Filter")>]
         let filter f (array: _[]) = 
-            checkNonNull "array" array
-            let res = List<_>() // ResizeArray
```

Comment:
@KevinRansom He's allocating an array of now instead of an array of 'T - so the bytes allocated could be significantly smaller, especially since you're always going to have to do a copy anyways at the end.